xarray/core/datatree_io.py

```diff
@@ -151,7 +151,7 @@ def _datatree_to_zarr(
         )
 
     for node in dt.subtree:
-        ds = node.ds
+        ds = node.to_dataset(local=True)
```

Comment:
I think [to_netcdf](https://github.com/shoyer/xarray/blob/datatree-hierarchy/xarray/core/datatree_io.py#L88) have the same treatment.